CONTRIBUTING.md

```diff
@@ -1,8 +1,74 @@
-# Contributing
+# Contributing to UMF (Unified Memory Framework)
+
+- [Opening new issues](#opening-new-issues)
+- [Submitting Pull Requests](#submitting-pull-requests)
+    - [Code style](#code-style)
+    - [License](#license)
+    - [Sign your work](#sign-your-work)
+    - [Adding new dependency](#adding-new-dependency)
+    - [Extending Public API](#extending-public-api)
+
+Below you'll find instructions on how to contribute to UMF, either with code changes
+or an issue with a problem to solve or a enhancement proposal.
+
+All contributions are most welcome!
+
+## Opening new issues
+
+Please log bugs or suggestions as [GitHub issues](https://github.com/oneapi-src/unified-memory-framework/issues).
+
+When reporting a bug remember about the details, at least:
+- version of UMF (hash of a commit),
+- OS and kernel,
+- compiler.
+
+Please use and follow instructions in the Issue template - its content depends on
+what type of issue you choose on opening.
+
+## Submitting Pull Requests
+
+We take outside code contributions to UMF through GitHub pull requests.
+
+<!--
+// copy-pasted from UR, to be reworded
+
+To submit a pull request to Unified Runtime, you must first create your own
+personal fork of the project and submit your changes to a branch. By convention
+we name our branches ``<your_name>/<short_description>``, where the description
+indicates the intent of your change. You can then raise a pull request
+targeting ``oneapi-src/unified-runtime:main``. Please add the *experimental*
+label to you pull request.
+
+When making changes to the specification you *must* commit all changes to files
+in the repository as a result of `Generating Source`_.
+
+Before your pull request is merged it *must* pass all jobs in the GitHub
+Actions workflow and *must* be reviewed by no less than two code owners.
+
+-->
+
+### Building and testing
+
+described in Readme...
+plus info about tests
+
+### Code style
+
+.clang-format
+UMF_FORMAT_CODE_STYLE
+make clang-format-check & apply
+DEVELOPER_MODE
+
+**NOTE**: We use specific clang-format - version exactly **15.0** is required.
+
+<!--
+### Naming Convention
```

Comment:
That's a good question ;d Originally we had the convention as in UR: camelCase for function names and snake_case for type names. However, I think now we also have some files where functions are named using snake_case - it would be nice to unify that.

Reply:
Please remove this section for the moment - I added a JIRA UMFW-248 for changing code to use single naming convention in all files + adding a section about it here to this doc